circuit-macros/src/circuit_type/mpc_types.rs

```diff
@@ -134,24 +121,37 @@ fn build_mpc_type(
 fn build_mpc_type_impl(mpc_type: &ItemStruct, base_type: &ItemStruct) -> TokenStream2 {
     let generics = base_type.generics.clone();
     let where_clause = generics.where_clause.clone();
-    let impl_generics = merge_generics(build_mpc_generics(), generics.clone());
 
-    let mpc_type_trait_name = with_mpc_generics(new_ident(MPC_ALLOC_TYPE_TRAIT_NAME));
-    let mpc_type_ident = ident_with_generics(mpc_type.ident.clone(), impl_generics.clone());
+    let mpc_type_trait_name = path_from_ident(new_ident(MPC_ALLOC_TYPE_TRAIT_NAME));
+    let mpc_type_ident = ident_with_generics(mpc_type.ident.clone(), generics.clone());
 
     // This ident is used for the `type NativeType` associated type
     let native_type_ident = new_ident(MPC_NATIVE_TYPE_ASSOCIATED_NAME);
-    let base_type_ident = ident_with_generics(base_type.ident.clone(), generics);
+    let base_type_ident = ident_with_generics(base_type.ident.clone(), generics.clone());
 
-    let authenticated_scalar_type =
-        ident_with_generics(new_ident(MPC_TYPE_SERIALIZED_IDENT), build_mpc_generics());
+    let authenticated_scalar_type = path_from_ident(new_ident(MPC_TYPE_SERIALIZED_IDENT));
     let from_auth_scalars_method = build_deserialize_method(
         new_ident(FROM_AUTHENTICATED_SCALARS_METHOD_NAME),
         authenticated_scalar_type.clone(),
         mpc_type_trait_name.clone(),
         mpc_type,
     );
 
+    // Build a `fabric` method
```

Comment:
hmm, relying on the ordering of fields feels a bit brittle, but idrk how you would more flexibly search for the first field that has a `fabric` method / implements some trait...

Reply:
We could eventually make this part of the trait that the mpc types implement, but that's more complexity than is needed for now, I'll add a todo